src/simulations/ensemblerun.jl

```diff
@@ -74,10 +78,11 @@ function series_ensemble(models, agent_step!, model_step!, n; kwargs...)
     return df_agent, df_model, models
 end
 
-function parallel_ensemble(models, agent_step!, model_step!, n; kwargs...)
+function parallel_ensemble(models, agent_step!, model_step!, n, batch_size; kwargs...)
     all_data = pmap(
         j -> run!(models[j], agent_step!, model_step!, n; kwargs...),
-        1:length(models),
+        1:length(models);
+        batch_size = batch_size
```

Comment:
```suggestion
        batch_size
```

Reply:
(This is exactly the reason we require julia 1.5+)

Reply:
My pleasure! Thank you for the excellent library. I am an avid Agents.jl user and evangelist and was hoping to find a way to contribute.

I agree, the explanation of `batch_size` was unclear in my message here and in the documentation. I have expanded the docstring with an example. I think it's good to include more detail since this is an important parameter that has yielded big performance gains for me. 

Basically `batch_size` should be increased when there are many more models to run than the number of processors available and each model run takes seconds or less to run. It's expensive for worker processors to communicate with the coordinating processor, so we should avoid making the processors do this excessively. Increasing batch size increases the number of models run by each worker before communicating, reducing communication overhead in the situation listed above. I know a bit, but I'm no expert on parallel computing; [this Julia parallel computing tutorial](https://techytok.com/lesson-parallel-computing) had a lot of helpful information for me (some [specifically about `pmap`](https://techytok.com/lesson-parallel-computing/#pmap), which I found via this helpful Discourse thread: https://discourse.julialang.org/t/parallelism-understanding-pmap-and-the-batch-size-parameter/15604.

I also fixed the `batch_size` kwarg redundancy you suggested with the latest push.

Reply:
If you want to contribute more, here is something that you can do with your expertise: add a new section in the Performance Tips page, that shows the impact of this parameter. Set up some real simple model where you can produce a plot of overall run time versus the `batch_size` keyword, and present the plot.

(if producing such a plot takes too much time, you can produce it and save it and then link it to the page instead)

Reply:
BTW, in the Performance Tips section you can have this more detailed description of the `batch_size` that you have here. It is just that my philosophy in documentation strings is that every sentence much be weighted in gold!

Reply:
No problem, I had a feeling it was too much. I updated the docs as you suggested.

I'd be happy to add a new section in Performance Tips. The Schelling model comes to mind as an example model, since that example uses `ensemblerun!`. Would a demonstration about different `batch_size` choices be useful in the Schelling example, or maybe just a one-sentence note about it in Schelling, with a link to the Performance Tips page?

Reply:
One sentence about it in Schelling, and linking to Performance Tips is a better approach I feel.

Reply:
fyi, still working on this, just doing some more tests and studying to make sure `batch_size` is the best way to accomplish this. I'm looking into approaches that automatically handle task batches sent to the workers. These seem to be greatly outperforming the current implicit default `batch_size = 1`, and could possibly save users from having to think about what batch size to set.

Reply:
thanks a lot, that is great!

Reply:
So I tried a different approach I think will work better and opened a new PR since it's a bit different solution: #624. Hope that's ok!